src/openassetio-python/cmodule/src/errors/BatchElementErrorBinding.cpp

```diff
@@ -32,7 +34,8 @@ void registerBatchElementError(const py::module& mod) {
       .def_readonly("code", &BatchElementError::code)
       .def_readonly("message", &BatchElementError::message)
       .def("__repr__", [](const BatchElementError& self) {
```

Comment:
[fix] I think this should be `__str__` and `__repr__` should be more-or-less what it was before, to be consistent with other types, and the idiom that `__repr__` is roughly copy-pasteable into a Python terminal.

Reply:
We discussed this during pairing. If we're going to do this, can you relax your insistance that the `str` formulation uses a colon rather than a comma. Then we can wrap the `str` one like we do in `EntityReference`, rather than having 2 different sources of truth for formatting.

Reply:
So `str` would be `errorCode, 'errorMessage'` and `repr` would be `BatchElementError(errorCode, 'errorMessage')`

I think that still doesn't work really for `repr`, since `errorCode` would be the stringified enum value, rather than the Python type, so is not a valid argument to the `BatchElementError` constructor.

We do have sort-of precedent in breaking that rule for `TraitsData`, since we can't currently construct a `TraitsData` from a Python dict. But I think we should be able to... we've discussed it before, since we've had to implement ad-hoc utility functions to convert between dict and TraitsData more than once (e.g. for BAL and simpleResolver).

Another thought is that ideally we'd use the `str` as the message in a `BatchElementException`, which would suite a `: ` better than `, ` separating the code from the description.  We don't currently include the code in the exception message, though, so not a strong argument.

Reply:
Just noticed `Context` is another example where the `repr` doesn't work as a copy-pasteable code. Argument is still valid, but much weaker, that a `Context` _should_ be constructible with a dict argument.

But that's probably a step too far. I think ideally `repr` for `Context` should be updated to match its constructor signature.

Reply:
Perhaps the quickest easiest solution is to just remove `__repr__` definitions from `BatchElementError` and `Context` (and perhaps elsewhere)?  So `str(...)` will still work and `repr(...)` will do whatever Python was already doing before.

Reply:
Removed the repr's.

https://github.com/OpenAssetIO/OpenAssetIO/pull/1329/commits/397559c0a76b9ff00cb7e18a4285d8e92bcf0a52